packages/core/EventManager.ts

```diff
@@ -255,8 +255,10 @@ export default class EventManager {
       results.push(result);
     }
 
-    // Update all calendar events.
-    results.push(...(await this.updateAllCalendarEvents(evt, booking)));
+    // If there is a calendar reference, update all calendar events.
+    if (booking.references.some((reference) => reference.type.includes("_calendar"))) {
+      results.push(...(await this.updateAllCalendarEvents(evt, booking)));
+    }
```

Comment:
I'm hoping this will also fix some rescheduling bugs. If there were no calendar references this was still throwing an error

Reply:
If `updateAllCalendarEvents` is supposed to have calendar references then instead of erroring shouldn't it ignore and early return? It seems hackish to do it outside of the fn when it's the functions requirement.

Reply:
Agreed with @hariombalhara. Let's find the root cause and fix it properly.

Reply:
@joeauyeung This one is still pending for merge.